spec/models/dependent/rules_spec.rb

```diff
@@ -0,0 +1,265 @@
+        let(:birth_date) { Date.new(tax_year - 40, 12, 25) }
+        it "is not a disqualified-child child relative" do
+          expect(subject.disqualified_child_qualified_relative?).to eq(false)
+        end
+      end
+    end
+  end
+
+  describe ".qualifying_relative?" do
+    context "with an old dependent who meets misc requirements and has a ssn/itin/atin stored" do
+      let(:meets_misc_qualifying_relative_requirements) { true }
+      let(:ssn_present) { true }
+
+      context "with a dependent who has a qualified child relationship but doesn't meet age conditions" do
+        let(:birth_date) { Date.new(tax_year - 70, 12, 25) }
+        let(:qualifying_child_relationship) { true }
+
+        it "returns true" do
+          expect(subject.qualifying_relative?).to eq true
+        end
+
+        context "when meet misc requirements are not met" do
+          let(:meets_misc_qualifying_relative_requirements) { false }
+
+          it "returns false" do
+            expect(subject.qualifying_relative?).to eq false
+          end
+        end
+      end
+
+      context "with a dependent who has a qualified relative relationship" do
+        let(:birth_date) { Date.new(tax_year - 30, 12, 25) }
+        let(:qualifying_relative_relationship) { true }
+
+        it "returns true" do
+          expect(subject.qualifying_relative?).to eq true
+        end
+
+        context "when meet misc requirements are not met" do
+          let(:meets_misc_qualifying_relative_requirements) { false }
+
+          it "returns false" do
+            expect(subject.qualifying_relative?).to eq false
+          end
+        end
+      end
+    end
+  end
+
+  describe ".meets_qc_residence_condition?" do
+    context "when already generally meeting the qualified child residence conditions" do
+      let(:meets_qc_residence_condition_generic) { true }
+
+      context "when younger than 6 months" do
+        let(:birth_date) { Date.new(tax_year, 12, 25) }
+        it "returns true" do
+          expect(subject.meets_qc_residence_condition?).to eq(true)
+        end
+      end
+
+      context "when older than 6 months" do
+        let(:birth_date) { Date.new(tax_year, 1, 1) }
+        it "returns true" do
+          expect(subject.meets_qc_residence_condition?).to eq(true)
+        end
+      end
+    end
+
+    context "when not generally meeting the qualified child residence conditions" do
+      let(:meets_qc_residence_condition_generic) { false }
+
+      context "when younger than 6 months" do
+        let(:birth_date) { Date.new(tax_year, 12, 25) }
+        it "returns true" do
+          expect(subject.meets_qc_residence_condition?).to eq(true)
+        end
+      end
+
+      context "when older than 6 months" do
+        let(:birth_date) { Date.new(tax_year, 1, 1) }
+        it "returns false" do
+          expect(subject.meets_qc_residence_condition?).to eq(false)
+        end
+      end
+    end
+  end
+
+  describe ".qualifying_child?" do
```

Comment:
i'm having trouble following this set of context blocks. they seem...random? the method basically just checks that everything in a list is true (the list consisting of some columns on dependent and some methods on rules) so i would have expected roughly:

```
context "when the child meets all of the qualifying conditions" do
end

context "when the child doesn't meet one or more of the qualifying conditions"do
end
```

...where each column / method was stubbed as true/false. the second block is trickier because it could feel excessively long to make one thing at a time false, but maybe something clever could be done looping over a list or something.
anyway, i don't get the age checks here, they seem to belong in `meets_qc_age_condition?`. i also realize this could be a holdover from the old code but maybe we should fix it